src/coreclr/src/jit/emit.cpp

```diff
@@ -5320,50 +5328,40 @@ UNATIVE_OFFSET emitter::emitFindOffset(insGroup* ig, unsigned insNum)
  *  block.
  */
 
-UNATIVE_OFFSET emitter::emitDataGenBeg(UNATIVE_OFFSET size, bool align)
+UNATIVE_OFFSET emitter::emitDataGenBeg(UNATIVE_OFFSET size, UNATIVE_OFFSET alignment)
 {
     unsigned     secOffs;
     dataSection* secDesc;
 
     assert(emitDataSecCur == nullptr);
 
-    /* The size better not be some kind of an odd thing */
+    // The size must not be zero and must be a multiple of 4
+    assert((size != 0) && ((size % 4) == 0));
```

Comment:
Interesting that the size is required to be a multiple of 4, but the alignment can be 1.

Reply:
The alignment can be requested to be 1/2 but will ultimately be at least 4 due to the minimum size requirement and the actual alignment of the data section.
It's simpler to allow the others, since they will ultimately be at least the requested alignment, than it is to block them.

Reply:
Modified the comment to:
> // The size must not be zero and must be a multiple of 4 bytes
    // Additionally, 4 bytes is the minimum alignment that will
    // actually be used. That is, if the user requests an alignment
    // of 1 or 2, they will get  something that is at least 4-byte
    // aligned. We allow the others since 4 is at least 1/2 and its
    // simpler to allow it than to check and block.